github/src/main/kotlin/gropius/sync/github/GithubSync.kt

```diff
@@ -0,0 +1,255 @@
+package gropius.sync.github
+
+import com.apollographql.apollo3.ApolloClient
+import gropius.model.architecture.IMSProject
+import gropius.model.issue.Issue
+import gropius.model.issue.Label
+import gropius.model.issue.timeline.IssueComment
+import gropius.model.template.IMSTemplate
+import gropius.model.template.IssueState
+import gropius.sync.*
+import gropius.sync.github.config.IMSConfigManager
+import gropius.sync.github.config.IMSProjectConfig
+import gropius.sync.github.generated.*
+import gropius.sync.github.generated.MutateAddLabelMutation.Data.AddLabelsToLabelable.Labelable.Companion.asIssue
+import gropius.sync.github.generated.MutateCreateCommentMutation.Data.AddComment.CommentEdge.Node.Companion.asIssueTimelineItems
+import gropius.sync.github.generated.MutateRemoveLabelMutation.Data.RemoveLabelsFromLabelable.Labelable.Companion.asIssue
+import gropius.sync.github.generated.fragment.TimelineItemData.Companion.asNode
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import java.net.URI
+
+/**
+ * This class is responsible for syncing data from and to GitHub
+ * @param githubDataService The service used to access the database
+ * @param issuePileService The service used to access the issue pile
+ * @param helper The helper used to convert data to and from JSON
+ * @param cursorResourceWalkerDataService The service used to access the cursor resource walker data
+ * @param imsConfigManager The manager used to access the IMS config
+ * @param collectedSyncInfo The collected sync info
+ * @param loadBalancedDataFetcher The load balanced data fetcher
+ */
+@Component
+final class GithubSync(
+    val githubDataService: GithubDataService,
+    val issuePileService: IssuePileService,
+    val helper: JsonHelper,
+    val cursorResourceWalkerDataService: CursorResourceWalkerDataService,
+    val imsConfigManager: IMSConfigManager,
+    collectedSyncInfo: CollectedSyncInfo,
+    val loadBalancedDataFetcher: LoadBalancedDataFetcher = LoadBalancedDataFetcher()
+) : AbstractSync(collectedSyncInfo), LoadBalancedDataFetcherImplementation, DataFetcher by loadBalancedDataFetcher {
+
+    init {
+        loadBalancedDataFetcher.start(this)
+    }
+
+    private val apolloClient = ApolloClient.Builder().serverUrl(URI("https://api.github.com/graphql").toString())
```

Comment:
hardcoded GitHub URL?

Reply:
the IMS with the settings is not available here. refactoring would not be wort it, as it is better done in one step when redoing netcode with users by login service

Reply:
oh dear